### PR TITLE
[PS4,PS5][Driver] Pass `-L<...>/target/lib -L.` to linker

### PR DESCRIPTION
**Pull request by playstation-edd**
The proprietary PS4 linker implicitly adds `=/target/lib` and `.` as library search paths. This behaviour was added to the PS5 linker via a downstream patch in LLD. This really belongs in the driver, instead. This change adds the driver behaviour to allow removal of the downstream patch in LLD.

There are no plans to update the PS4 linker behaviour in the analogous way, so do not pass the same search paths to the PS4 linker.

SIE tracker: TOOLCHAIN-16704

### CODE REVIEW

File: clang/test/Driver/ps5-linker.c [hunk 1]
```diff
@@ -46,3 +46,26 @@
...
+// Test that <sdk-root>/target/lib is added to library search paths, if it
+// exists and no --sysroot is specified.
+
+// RUN: rm -rf %t.dir && mkdir -p %t.dir/target/lib
+// RUN: env SCE_PROSPERO_SDK_DIR=%t.dir %clang --target=x64_64-sie-ps5 %s -### 2>&1 | FileCheck --check-prefixes=CHECK-TARGETLIB %s
+
+// CHECK-TARGETLIB: {{ld(\.exe)?}}"
+// CHECK-TARGETLIB-SAME: "-L{{.*[/\\]}}target/lib"
+
+// RUN: env SCE_PROSPERO_SDK_DIR=missing %clang --target=x64_64-sie-ps5 %s -### 2>&1 | FileCheck --check-prefixes=CHECK-NO-TARGETLIB %s
```
Comment (pogo59) [inline clang/test/Driver/ps5-linker.c:67]: This assumes `missing` is not present. While that's likely to be true, I might prefer to sequence the tests this way:
```
rm -rf %t.dir && mkdir -p %t.dir
env SCE_PROSPERO_SDK_DIR=%t.dir .... CHECK-NO-TARGETLIB
mkdir -p %t.dir/target/lib
env SCE_PROSPERO_SDK_DIR=%t.dir .... CHECK-TARGETLIB
```
It's a smidge more robust and self-explanatory IMO.

File: clang/test/Driver/ps5-linker.c [hunk 2]
```diff
@@ -46,3 +46,26 @@
...
+// Test that "." is always added to library search paths. This is long-standing
+// behavior, unique to PlayStation toolchains.
+
+// RUN: %clang --target=x64_64-sie-ps5 %s -### 2>&1 | FileCheck --check-prefixes=CHECK-LDOT %s
+
+// CHECK-LDOT: {{ld(\.exe)?}}"
+// CHECK-LDOT-SAME: "-L."
+
+// Test that <sdk-root>/target/lib is added to library search paths, if it
+// exists and no --sysroot is specified.
```
Comment (pogo59) [inline clang/test/Driver/ps5-linker.c:59]: Demonstrate that --sysroot causes no `target/lib` to be appended? Or is that tested elsewhere?

File: clang/test/Driver/ps5-linker.c (hunk 1 above)
Comment (playstation-edd) [reply clang/test/Driver/ps5-linker.c]: Yes, much better. Done. Thanks.

File: clang/test/Driver/ps5-linker.c (hunk 2 above)
Comment (playstation-edd) [reply clang/test/Driver/ps5-linker.c:59]: This is what the RUN on line 63 does ... I hope! (Prior to 1a6dc7d, this was on line 68).

Comment (pogo59) [reply clang/test/Driver/ps5-linker.c:59]: Ah missed that, sorry!

**Review by pogo59 — APPROVED** [review]
LGTM

**Pull request merged**